### ISSUE TITLE
How do I put default values in multiselect 

### ISSUE THREAD

**Issue by marisoru** [truckingsim/Ajax-Bootstrap-Select#41]
I need to include selected values by default. What I mean is to put pre selected values when I initialize the ajax bootstrap select. I tried using .selectpicker('val',[]) and it puts the values selected but not with the ajax bootstrap select styles, and when I want to add a new value typing on the search input the preselected values are deleted.

**Comment by truckingsim** [comment]
Can you post the ajaxSelectPicker code you are using? So we can see your config.

**Comment by marisoru** [comment]
The options:

 var optionsGroups = {
                ajax: {
                    url: 'google.com',
                    type: 'GET',
                    dataType: 'json',
                    data: {
                        term: '{{{q}}}'
                    }
                },
                log: 3,
                preprocessData: function (data) {
                    var i, l = data.length, array = [];
                    if (heart) {
                        for(i = 0; i < l; i++){
                            array.push($.extend(true, data[i], {
                                text: data[i].name,
                                value: data[i].id,
                                data: {
                                    subtext: data[i].subText
                                }
                            }));
                        }
                    }
                    return array;
                }
            }

The select 

<select id="reportTemplateGroupFieldValues" class="selectpicker form-control groupAjax" name="reportTemplateGroupValues" multiple data-live-search="true" data-selected-text-format="count" title="Choose..." data-size="200px" data-count-selected-text="{0} groups selected">
            
                <option value="1" selected>Brasil</option>
				<option value="2" selected>Colombia</option>
            
</select>

and 

$('#reportTemplateGroupFieldValues').selectpicker().ajaxSelectPicker(optionsGroups);

**Comment by truckingsim** [comment]
So we have an option called `preserveSelected` which will keep previously selected options between ajax requests.  Try adding that to your options hash.

```js
var optionsGroups = {
  ajax: { ... },
  log: 3,
  preprocessData: function(){ ... },
  preserveSelected: true
};

**Comment by diegoaam** [comment]
Hello, and thanks for the quick answers you give, I have the same problem, I tried with the preserveSelected that you said, and it does not work, I think the problem is that the style of ajaxSelectPicker are not applied, and when I type something it applies the styles and for sure deletes the selected values

![showcase](https://cloud.githubusercontent.com/assets/753830/6373553/4a7f103e-bcea-11e4-8c8d-c15706ca9af0.png)

**Comment by truckingsim** [comment]
Ahh that second image helps.  Ok yeah this is a bug.  Basically we aren't checking for currently selected on initialization only on update/keydown.  It will probably be this weekend before I  have time to work on this though.

**Comment by Allerion** [comment]
+1, I also need this for my system. Thanks

**Comment by truckingsim** [comment]
@diegoaam @markcarver I've spent 5 hours on this so far over multiple days and haven't been able to find out why it can't keep initial selected options yet.  

@markcarver If you can find time, I'm pushing up a bugfix branch which has what I've done so far, maybe you'll see something.  The commit df4b2b5 has the change I made that will save one of the options but not both.  I also added a 4th select to the example page for testing this.

**Comment by markcarver** [comment]
I'll have to look at this in more depth later. This month is going to be chaotic for me (family/moving).

**Comment by truckingsim** [comment]
Thanks for giving it a quick once over.  Just moved myself so fully understand.  Good luck!

**Comment by rockstarreynolds** [comment]
Hi there - I've downloaded the latest build and I don't see that this issue is fixed?

**Comment by truckingsim** [comment]
Adding this 2.0.0 I really want to get this fixed.

**Comment by jengjeng** [comment]
You need to insert `<option selected>` tag before initializing select picker and prevent cache bug by using this code
`// Use after init select picker`
`$('#select-picker').data('AjaxBootstrapSelect').list.cache = {}`

You can try from this fiddle https://jsfiddle.net/a3ybhne4/9/

**Comment by callumstrubi** [comment]
Can confirm that the fix from @jengjeng works and also fixes a secondary bug of de-selecting pre-selected options that must be incorrectly initialised.

I'd consider this bug critical and without the workaround we'd have had to implement an alternative library entirely.

**Comment by markhalliwell** [comment]
I believe this will be fixed by #164
